lib/tapioca/compilers/dsl/active_record_relations.rb

```diff
@@ -219,13 +219,19 @@ def generate
             query_methods |= ActiveRecord::SpawnMethods.instance_methods(false)
             # Remove the ones we know are private API
             query_methods -= [:arel, :build_subquery, :construct_join_dependency, :extensions, :spawn]
+            # Remove "where" which needs a custom return type for WhereChains
+            query_methods -= [:where]
             # Remove the methods that ...
             query_methods
               .grep_v(/_clause$/) # end with "_clause"
               .grep_v(/_values?$/) # end with "_value" or "_values"
               .grep_v(/=$/) # end with "=""
               .grep_v(/(?<!uniq)!$/) # end with "!" except for "uniq!"
           end, T::Array[Symbol])
+          WHERE_CHAIN_QUERY_METHODS = T.let(
+            ActiveRecord::QueryMethods::WhereChain.instance_methods(false),
```

Comment:
While we don't have tests for it, we do try to keep support for Rails 5.2 as much as we  can. If this is a class added in Rails 6, maybe we should check that it is defined first?